allPostsToWordpress.py

```diff
@@ -21,7 +21,12 @@ def addPost(post, channel, username, wpUrl, doProtected, protectedPassword, doCo
     ET.SubElement(item, 'wp:post_date_gmt').text = pubDate.astimezone(datetime.timezone.utc).strftime('%Y-%m-%d %H:%M:%S')
     ET.SubElement(item, 'wp:comment_status').text = 'open'
     ET.SubElement(item, 'wp:ping_status').text = 'open'
-    ET.SubElement(item, 'wp:post_name').text = post.get('linkId')
+    postTitle = post.find('title').text.strip().lower()
+    wordChar = re.sub('[^\s\w]', '', postTitle)
+    if postTitle == '(no subject)':
+        ET.SubElement(item, 'wp:post_name').text = post.get('linkId')
```

Comment:
Ah, you have the right idea here.  Here's what I was thinking:

    if safeTitle == '(no subject)':
        postName = post.get('linkId')
    else:
        postName = consolidateJoinedSpaces(spacesToDashes(wordChar))
    ET.SubElement(item, 'wp:post_name').text = postName

(although see my comment below about the else clause)

Does that make sense?